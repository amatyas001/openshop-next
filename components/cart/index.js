export * from './CartContent/CartContent';
export * from './CartAmount/CartAmount';
export * from './CartButton/CartButton';
export * from './CartItem/CartItem';
export * from './CartPanel/CartPanel';
