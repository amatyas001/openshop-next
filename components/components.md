Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
