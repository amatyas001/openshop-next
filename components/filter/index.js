export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
