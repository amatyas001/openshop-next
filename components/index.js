export * from './cart';
export * from './filter';
export * from './product';
export * from './layout';
export * from './payment';
export * from './shared';
