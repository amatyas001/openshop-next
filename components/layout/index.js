export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
