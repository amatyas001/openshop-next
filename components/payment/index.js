export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
