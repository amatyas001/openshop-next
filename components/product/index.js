export * from './ProductCard/ProductCard';
export * from './ProductControls/ProductControls';
export * from './ProductControlsColor/ProductControlsColor';
export * from './ProductControlsAmount/ProductControlsAmount';
export * from './ProductControlsButton/ProductControlsButton';
export * from './ProductControlsSize/ProductControlsSize';
