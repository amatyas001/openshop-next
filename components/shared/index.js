export * from './Button/Button';
export * from './ButtonIcon/ButtonIcon';
export * from './Field/Field';
export * from './Image/Image';
export * from './Panel/Panel';
export * from './Spinner/Spinner';
