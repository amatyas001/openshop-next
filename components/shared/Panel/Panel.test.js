import { create, act } from 'react-test-renderer';
import { Panel } from '@app/components';

describe('<Panel />', () => {
  let tree;

  beforeAll(() => {
    act(() => {
      tree = create(<Panel />);
    });
  });

  it('should render without props', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
