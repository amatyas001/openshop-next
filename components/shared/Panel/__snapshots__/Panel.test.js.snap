// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> should render without props 1`] = `
<div
  aria-hidden={true}
  className="rah-static rah-static--height-zero css-vppuuz"
  style={
    Object {
      "height": 0,
      "overflow": "hidden",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTransition": "opacity 250ms ease 0ms",
        "opacity": 0,
        "transition": "opacity 250ms ease 0ms",
      }
    }
  >
    <div
      className="css-1wfrc3n"
    />
  </div>
</div>
`;
